lib/mailpy.rb

```diff
@@ -20,9 +20,19 @@ def deliver!(mail)
     def perform_send_request(mail, settings)
       begin
         result = MailerApi.new(mail, settings).send
-        raise(MailpyDeliveryError, JSON.parse(result.body)['message']) unless result.code === 200
+    
+        unless result.code === 200
+          message = begin
+            JSON.parse(result.body)['message']
+          rescue JSON::ParserError
+            Rails.logger.error("Mailpy response is not JSON: #{result.body[0..300]}")
+            "Unexpected response from Mailpy: #{result.body[0..100]}"
+          end
+          raise(MailpyDeliveryError, message)
+        end
```

Comment:
It might be cleaner to extract this block into a private function. Let's call it `extract_error_message`.




```suggestion
        # we return early here if the response code is 200
        return result if result.code == 200
      
        # otherwise, raise an error
        error_message = extract_error_message(result.body)
        raise(MailpyDeliveryError, error_message)
```

Reply:
well noted, Nasrul.
thanks for the kind advice.